src/xfieldhistory.cpp

```diff
@@ -7,12 +7,15 @@
 #include <txdb.h>
 #include <univalue.h>
 #include <sync.h>
+#include <shared_mutex>
 
 XFieldHistoryMapType CXFieldHistoryMap::xfieldHistory;
-std::mutex CXFieldHistoryMap::xfieldHistoryMutex;
+std::shared_mutex CXFieldHistoryMap::xfieldHistoryMutex;
 
+// Internal helper without lock - caller must hold lock
 bool CXFieldHistoryMap::IsNew(TAPYRUS_XFIELDTYPES type, const XFieldChange& xFieldChange) const
 {
+    std::shared_lock<std::shared_mutex> lock(xfieldHistoryMutex);
```

Comment:
The comment says "Internal helper without lock - caller must hold lock", but in fact it acquires a shared_lock.